ng-dev/auth/shared/node-handler.ts

```diff
@@ -0,0 +1,144 @@
+import {EventEmitter} from 'events';
+import * as Http from 'http';
+import * as Url from 'url';
+
+import {AuthorizationRequest} from '@openid/appauth';
+import {AuthorizationRequestHandler, AuthorizationRequestResponse} from '@openid/appauth';
+import {AuthorizationError, AuthorizationResponse} from '@openid/appauth';
+import {AuthorizationServiceConfiguration} from '@openid/appauth';
+import {Crypto} from '@openid/appauth';
+import {log} from '@openid/appauth';
+import {BasicQueryStringUtils, QueryStringUtils} from '@openid/appauth';
+import {NodeCrypto} from '@openid/appauth/built/node_support/crypto_utils.js';
+
+import opener from 'opener';
+
+class ServerEventsEmitter extends EventEmitter {
+  static ON_UNABLE_TO_START = 'unable_to_start';
+  static ON_AUTHORIZATION_RESPONSE = 'authorization_response';
+}
+
+export class NodeBasedHandler extends AuthorizationRequestHandler {
+  // the handle to the current authorization request
+  authorizationPromise: Promise<AuthorizationRequestResponse | null> | null = null;
+
+  /** The content for the authorization redirect response page. */
+  protected authorizationRedirectPageContent =
+    'Close your browser to continue<script>window.close()</script>';
+
+  constructor(
+    // default to port 8000
+    public httpServerPort = 8000,
+    utils: QueryStringUtils = new BasicQueryStringUtils(),
+    crypto: Crypto = new NodeCrypto(),
+  ) {
+    super(utils, crypto);
+  }
+
+  performAuthorizationRequest(
+    configuration: AuthorizationServiceConfiguration,
+    request: AuthorizationRequest,
+  ) {
+    // use opener to launch a web browser and start the authorization flow.
+    // start a web server to handle the authorization response.
+    const emitter = new ServerEventsEmitter();
+
+    const requestHandler = (httpRequest: Http.IncomingMessage, response: Http.ServerResponse) => {
+      if (!httpRequest.url) {
+        return;
+      }
+
+      const url = Url.parse(httpRequest.url);
+      const searchParams = new Url.URLSearchParams(url.query || '');
+
+      const state = searchParams.get('state') || undefined;
+      const code = searchParams.get('code');
+      const error = searchParams.get('error');
+
+      if (!state && !code && !error) {
+        // ignore irrelevant requests (e.g. favicon.ico)
+        return;
+      }
+
+      log('Handling Authorization Request ', searchParams, state, code, error);
+      let authorizationResponse: AuthorizationResponse | null = null;
+      let authorizationError: AuthorizationError | null = null;
+      if (error) {
+        log('error');
```

Comment:
Why do we use the logging functions from openid and not the dev-infra ones we have? with the nice error file. Specifically relevant for the error logging below in the `.catch`